src/Component.php

```diff
@@ -5,12 +5,155 @@
+            throw new UserException(
+                'Manifest file for table ' . $fileName . ' does not specify delimiter.'
+            );
+        }
+        if (!isset($manifest['enclosure'])) {
+            throw new UserException(
+                'Manifest file for table ' . $fileName . ' does not specify enclosure.'
+            );
+        }
+    }
+
+    private function getMaxColCount(SplFileInfo $file, string $delimiter, string $enclosure) : int
+    {
+        $csvFile = new CsvFile($file->getPathname(), $delimiter, $enclosure);
+        $maxColCount = 0;
+        foreach ($csvFile as $row) {
+            $maxColCount = max($maxColCount, count($row));
+        }
+        return $maxColCount;
+    }
+
+    private function fillHeader(array $header, int $maxColCount) : array
+    {
+        for ($i = 0; $i < $maxColCount; $i++) {
+            if (empty($header[$i])) {
+                $header[$i] = 'auto_col_' . $i;
+            }
+        }
+        return $header;
+    }
+
+    private function orthogonalize(
+        SplFileInfo $file,
+        string $destination,
+        int $maxColCount,
+        string $delimiter,
+        string $enclosure
+    ) : void {
+        $sourceCsv = new CsvFile($file->getPathname(), $delimiter, $enclosure);
+        $destinationCsv = new CsvFile($destination, $delimiter, $enclosure);
+        foreach ($sourceCsv as $index => $row) {
+            $destinationCsv->writeRow(array_pad($row, $maxColCount, ''));
+        }
+    }
+
+    private function writeManifest(SplFileInfo $file, array $manifest, array $header) : void
+    {
+        $manifest["columns"][] = $header;
+        $targetManifest = $this->getDataDir() . '/out/tables/' . $file->getFilename() . ".manifest";
+        file_put_contents(
+            $targetManifest,
+            $this->jsonEncode->encode($manifest, JsonEncoder::FORMAT)
+        );
+    }
+
+    private function readManifest(SplFileInfo $file) : array
+    {
+        $manifestFile = $file->getPathname() . '.manifest';
+        if (!$this->fs->exists($manifestFile)) {
+            throw new UserException(
+                'Table ' . $file->getBasename() . ' does not have a manifest file.'
+            );
+        }
+
+        $manifest = $this->jsonDecode->decode(file_get_contents($manifestFile), JsonEncoder::FORMAT);
+        $this->validateManifest($manifest, $file->getFilename());
+        return $manifest;
+    }
+
     public function run(): void
     {
-        // @TODO implement
+        $finder = new Finder();
+        $finder->in($this->getDataDir() . '/in/tables')->notName('*.manifest')->depth(0);
+        foreach ($finder as $file) {
+            $manifest = $this->readManifest($file);
+            $header = $manifest['columns'];
+            $delimiter = $manifest['delimiter'];
+            $enclosure = $manifest['enclosure'];
+
+            if (is_dir($file->getPathname())) {
+                // sliced file
+                $finder = new Finder();
```

Comment:
přepíše se vnějsí `$finder` !!!

Reply:
no neprepise, protoze ve foreachu to nejde, ale prejmenoval jsem